src/platforms/share/Entity.ts

```diff
@@ -0,0 +1,279 @@
+import { Response } from "ask-sdk-model";
+import * as _ from "lodash";
+import { AlexaReply, DialogflowReply, ITransition, IVoxaReply } from "../..";
+import { IDirective } from "../../directives";
+import { IVoxaEvent } from "../../VoxaEvent";
+
+export enum EntityOverrideMode {
+  Unspecified = "ENTITY_OVERRIDE_MODE_UNSPECIFIED",
+  Override = "ENTITY_OVERRIDE_MODE_OVERRIDE",
+  Supplement = "ENTITY_OVERRIDE_MODE_SUPPLEMENT",
+}
+
+export enum UpdateBehavior {
+  Replace = "REPLACE",
+  Clear = "CLEAR",
+}
+
+export abstract class EntityHelper {
+  public getEntity(rawEntity: any, event: IVoxaEvent): any {
+    const platform =
+      _.get(event, "platform.name") ||
+      _.get(event, "rawEvent.originalDetectIntentRequest.source");
+
+    let entity: any;
+    let behavior: any;
+
+    ({ entity, behavior } = this.generateEntityFormat(
+      rawEntity,
+      platform,
+      event,
+    ));
+
+    if (platform === "alexa") {
```

Comment:
it should be possible to compose our code have this checks in the platform specific functions and not in the share code that's supposed to be platform independent